src/bin/pg_rewind/pg_rewind.c

```diff
@@ -490,6 +491,30 @@ main(int argc, char **argv)
 	return 0;
 }
 
+file_action_t
+decide_wal_file_action(const char fname[MAXFNAMELEN])
+{
+    TimeLineID  file_tli;
+    XLogSegNo	file_segno;
+    XLogSegNo   last_common_segno;
+
+    /*
+	 * Find last common WAL segment number between source and target before
+	 * divergence given last common LSN (byte position).
+     *
+     * TODO: ensure this reacts to different WAL segment sizes
```

Comment:
Will add a test case that will set a different WAL segment size instead of the standard 16MB, and then run `pg_rewind`.